src/main/java/com/robohorse/robopojogenerator/generator/postprocessors/KotlinDataClassPostProcessor.java

```diff
@@ -0,0 +1,183 @@
+            classBodyBuilder.deleteCharAt(classBodyBuilder.lastIndexOf(","));
+        }
+
+        return classBodyBuilder.toString();
+    }
+
+
+    private String proceedField(String objectName) {
+
+        String fieldName = objectName;
+
+        if (fieldName.contains("-")) {
+            // Turn to lower case with underscore if it is hyphen
+            fieldName = fieldName.replaceAll("-+", "_");
+        }
+
+        if (fieldName.contains("_")) {
+            fieldName = fieldName.replaceAll("_{2,}", "_");
+            fieldName = CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, fieldName);
+        }
+
+        char fieldNameFirstChar = fieldName.charAt(0);
+        if (Character.isDigit(fieldNameFirstChar)) {
+            // The first char is number
+            fieldName = "_" + fieldName;
+        }
+
+        return fieldName;
+    }
+
+
+    @Override
+    protected String createClassItemText(String packagePath, String imports, String classTemplate) {
+
+        return classTemplateProcessor.createClassItemWithoutSemicolon(packagePath,
+                                                                      imports,
+                                                                      classTemplate);
+    }
+
+
+    private String proceedAnnotation(String annotation) {
+
+        if (annotation != null) {
+            String regex = "\"@(.+?)\"";
+            annotation = annotation.replaceAll(regex, "\"ç$1\"");
+            annotation = annotation.replaceAll("@", "@field:");
+
+            regex = "\"ç(.+?)\"";
+            annotation = annotation.replaceAll(regex, "\"@$1\"");
+
+            return annotation;
+        }
+        return null;
+    }
+
+
+    private String proceedType(String type) {
+
+        final String  regex   = "<([^<>]*)>";
+        final Pattern pattern = Pattern.compile(regex);
+        final Matcher matcher = pattern.matcher(type);
+
+        String innerType;
+        if (matcher.find()) {
+            // Type is List
+
+            innerType = generateHelper.upperCaseFirst(matcher.group(1));
+
+            if (innerType.equals(ClassType.OBJECT.getBoxed())) {
+                innerType = "Any";
+            }
+            else if (innerType.equals(ClassType.INTEGER.getBoxed())) {
+                innerType = "Int";
+            }
+
+            type = type.replaceAll(regex, "<" + innerType + ">");
+            type = type.replaceAll(">", "?>");
+        }
+        else {
+            // Type is object or primitive
+
+            type = generateHelper.upperCaseFirst(type);
+
+            if (type.equals("Object")) {
+                type = "Any";
+            }
+        }
+
+        return type;
+    }
+
+
+    /**
+     * Remove List import and semicolon
+     *
+     * @param classItem The class item which is use to getting import
+     */
+    private void proceedImports(ClassItem classItem) {
+
+        Set<String> classImports = classItem.getClassImports();
```

Comment:
What about make a processor factory?
Just like the post processor factory.

It will gain more flexibility with the class processing.

-----

I try to add a processor factory, and the plugin works well.

But there is a test fail.
I don't know how to fix it.